tests/e2e/e2e_utils.py

```diff
@@ -43,16 +48,27 @@ def run_oneshot_for_e2e_testing(
     # Load model.
     oneshot_kwargs = {}
 
-    loaded_model, tokenizer = _load_model_and_tokenizer(model=model, device=device)
+    loaded_model, processor = _load_model_and_processor(
+        model=model, model_class=model_class, device=device
+    )
 
     if dataset_id:
         ds = load_dataset(dataset_id, name=dataset_config, split=dataset_split)
         ds = ds.shuffle(seed=42).select(range(num_calibration_samples))
-        ds = preprocess_tokenize_dataset(ds, tokenizer, max_seq_length)
+        ds = process_dataset(ds, processor, max_seq_length)
         oneshot_kwargs["dataset"] = ds
         oneshot_kwargs["max_seq_length"] = max_seq_length
         oneshot_kwargs["num_calibration_samples"] = num_calibration_samples
 
+        # Define a data collator for multimodal inputs.
+        if "flickr30k" in dataset_id:
+
+            def data_collator(batch):
+                assert len(batch) == 1
```

Comment:
why does the batch need to be 1 here? 
memory constraint? Could you add the error message also

Reply:
ah! good eye, that should be `>=1`, my bad

Reply:
@brian-dellabetta the input to the data collator is a list whose length is the batch size. This collator indeed only works for batch_size = 1 because we haven't yet developed a universal multimodal padding function, although such a thing exists in vllm

Reply:
@kylesayrs ah ok, should I use `batch size=1` in lmeval then? I've got it set to 4 or 8 usually, it must work because they also have a compatible padding function?
where does batch size even get set in the call to oneshot? I don't see it on `DatasetArguments` or `TrainingArguments`

Reply:
@brian-dellabetta No, the data collator defined here is only passed into oneshot, not eval

Reply:
Right, does that mean they also have this data collation function? wondering how hard it would be to pull it, as part of a separate issue from this

Reply:
Marking this as resolved, @kylesayrs let's chat in our 1:1 about it as a potential next step though?